If libncurses is configured to build libtinfo as a separate library,
part of the symbols that used to be in libncurses are now in libtinfo

Use ncurses.pc to get correct "Libs:" and "Libs.private:"

http://bugs.gentoo.org/454322

--- configure.in
+++ configure.in
@@ -374,7 +374,7 @@
 AC_CHECK_HEADERS(ncurses/termcap.h)
 AC_CHECK_LIB(termcap, initscr, HAVE_TERMCAP="termcap")
 AC_CHECK_LIB(curses, initscr, HAVE_TERMCAP="curses")
-AC_CHECK_LIB(ncurses, initscr, HAVE_TERMCAP="ncurses")
+PKG_CHECK_MODULES(TERMCAP, ncurses, HAVE_TERMCAP="ncurses", HAVE_TERMCAP="")
 
 AM_ICONV
  
@@ -621,7 +621,7 @@
 dnl todo: always use yes as default, use simulation instead ?
 AC_MSG_CHECKING(for termcap)
 if test "x${HAVE_TERMCAP}" != "x"; then
-  FRONTEND_LDADD="-l${HAVE_TERMCAP} ${FRONTEND_LDADD}"
+  FRONTEND_LDADD="-l${HAVE_TERMCAP} ${TERMCAP_LIBS} ${FRONTEND_LDADD}"
   AC_DEFINE(HAVE_TERMCAP, 1, have termcap)
   TERMCAP_DEFAULT="yes"
 else
